search/src/main/nodejs/search_combiner.js

```diff
@@ -269,40 +269,28 @@ export const searchCombiner = async (event) => {
     if (DEBUG) console.log('Input event:', JSON.stringify(event));
 
     // Parameters
-    const { jobId, tasksTableName, timedOut, completed, withErrors, fatalErrors } = event;
+    const { jobId, tasksTableName, timedOut, completed, batchErrors } = event;
     const { searchBucket, searchId, maskKeys, maxResultsPerMask } = event.jobParameters;
     const fullSearchInputName = maskKeys[0];
     const searchInputName = fullSearchInputName.substring(fullSearchInputName.lastIndexOf("/") + 1);
 
     let allBatchResults = {};
 
     const now = new Date();
-    if (fatalErrors && fatalErrors.length > 0) {
-        // if there are fatalErrors do not try any reducing step because
-        // the process had already been invoked and most likely failed
-        // so simply report the error just in case it has not been reported yet
-        console.log(`Job ${jobId} - ${searchId} completed with fatal errors`, fatalErrors);
-        await updateSearchMetadata({
-            id: searchId,
-            step: SEARCH_COMPLETED,
-            errorMessage: "Color depth search completed with fatal errors",
-            cdsFinished: now.toISOString()
-        });
-        return event;
-    } else if (timedOut) {
+    if (timedOut || !completed) {
         console.log(`Job ${jobId} - ${searchId} timed out`);
         await updateSearchMetadata({
             id: searchId,
             step: SEARCH_COMPLETED,
             errorMessage: "Color depth search timed out",
             cdsFinished: now.toISOString()
         });
-    } else if (withErrors || !completed) {
-        console.log(`Job ${jobId} - ${searchId} completed with errors`);
+    } else if (batchErrors && batchErrors.length > 0) {
+        console.log(`Job ${jobId} - ${searchId} completed with errors`, batchErrors);
         await updateSearchMetadata({
             id: searchId,
             step: SEARCH_COMPLETED,
-            errorMessage: "Color depth search completed with errors",
+            errorMessage: `Color depth search completed with errors: ${JSON.stringify(batchErrors)}`,
```

Comment:
What is in the batchErrors variable? Is it useful for the end user to see it? Should we just log it and report a generic error message, and tell them to contact us?

Reply:
What do you think is best from a UX perspective? Finding the error in the cloudwatch logs is a nightmare, but they are also in the step function execution. Some errors can be confusing for users so a simple generic message may be better.

Reply:
If you want to surface the error on the client for debugging, then we should package it in a way that allows us to log it to the console or display it if an admin is logged in.